src/canopy_cli/data_loader/data_loader.py

```diff
@@ -67,30 +72,107 @@ def _df_to_documents(df: pd.DataFrame) -> List[Document]:
     return documents
 
 
-def _load_single_file_by_suffix(file_path: str) -> List[Document]:
+def _load_multiple_txt_files(file_paths: List[str]) -> pd.DataFrame:
+    """Load multiple text files into a single dataframe
+
+    Args:
+        file_paths: List of file paths to load
+
+    Returns:
+        pd.DataFrame: Dataframe with columns `id`, `text` and 'source`
+                      Note: metadata will be empty
+    """
+    if not isinstance(file_paths, list):
+        raise ValueError("file_paths must be a list of strings")
+    if len(file_paths) == 0:
+        raise ValueError("file_paths must not be empty")
+
+    df = pd.DataFrame(columns=["id", "text", "source"])
+    rows = []
+    for file_path in file_paths:
+        with open(file_path, "r") as f:
+            text = f.read()
+            rows.append(
+                {
+                    "id": os.path.basename(file_path),
+                    "text": text,
+                    "source": file_path
+                }
+            )
+        df = pd.DataFrame(rows, columns=["id", "text", "source"])
```

Comment:
BUG!!

This line should be outside of the `for` loop. You're creating a new `df` in every loop iteration